### ISSUE TITLE
cannot push to local registry

### ISSUE THREAD

**Issue by Baykonur** [pipo02mix/why_k8s_can_make_our_life_easier#13]
I have a virtual k8s:
```
$ kubectl get nodes -o wide
NAME         STATUS   ROLES    AGE    VERSION   INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                KERNEL-VERSION               CONTAINER-RUNTIME
master.k8s   Ready    master   3d2h   v1.9.8    192.168.99.104   <none>        CentOS Linux 7 (Core)   3.10.0-862.11.6.el7.x86_64   docker://1.13.1
node1.k8s    Ready    <none>   3d2h   v1.9.8    192.168.99.107   <none>        CentOS Linux 7 (Core)   3.10.0-862.11.6.el7.x86_64   docker://1.13.1
node2.k8s    Ready    <none>   3d2h   v1.9.8    192.168.99.106   <none>        CentOS Linux 7 (Core)   3.10.0-862.11.6.el7.x86_64   docker://1.13.1
node3.k8s    Ready    <none>   3d2h   v1.9.8    192.168.99.105   <none>        CentOS Linux 7 (Core)   3.10.0-862.11.6.el7.x86_64   docker://1.13.1

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T17:05:32Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.11", GitCommit:"1bfeeb6f212135a22dc787b73e1980e5bccef13d", GitTreeState:"clean", BuildDate:"2018-09-28T21:35:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Those are the steps I do on my virtual k8s cluster:
```
$ kubectl get pods --namespace kube-system -l k8s-app=kube-registry-upstream
NAME                                READY   STATUS    RESTARTS   AGE
kube-registry-v0-598b5746b7-f7pp2   1/1     Running   0          30m
$ kubectl port-forward -n kube-system kube-registry-v0-598b5746b7-f7pp2 5000
Forwarding from 127.0.0.1:5000 -> 5000
Forwarding from [::1]:5000 -> 5000
```

Added `localhost:5000` as insecure docker registry at Docker-> preferences and restarted Docker.

On a separate shell I tagged and pushed busybox as instructed:
```
$ docker pull busybox
Using default tag: latest
latest: Pulling from library/busybox
Digest: sha256:2a03a6059f21e150ae84b0973863609494aad70f0a80eaeb64bddd8d92465812
Status: Image is up to date for busybox:latest
$ docker tag busybox localhost:5000/busybox:latest
$ docker push localhost:5000/busybox:latest
```
And below is the error I get:
```
The push refers to repository [localhost:5000/busybox]
Get http://localhost:5000/v2/: dial tcp [::1]:5000: connect: connection refused
```

and the logs of the kube-registry-v0
```
time="2018-10-31T22:00:47Z" level=info msg="response completed" go.version=go1.7.6 http.request.host="localhost:5000" http.request.id=bd3e2dd3-6700-46bf-9860-f1b054302b37 http.request.method=GET http.request.remoteaddr="127.0.0.1:60678" http.request.uri="/v2" http.request.useragent="curl/7.54.0" http.response.duration="172.108µs" http.response.status=301 http.response.written=39 instance.id=d8099907-66af-4c26-8797-e7bf994d3bc7 version=v2.6.2 
127.0.0.1 - - [31/Oct/2018:22:00:47 +0000] "GET /v2 HTTP/1.1" 301 39 "" "curl/7.54.0"
time="2018-10-31T22:01:03Z" level=info msg="response completed" go.version=go1.7.6 http.request.host="localhost:5000" http.request.id=66b0c39b-538d-461c-b66b-bb46b9f3c070 http.request.method=GET http.request.remoteaddr="127.0.0.1:60716" http.request.uri="/v2" http.request.useragent="curl/7.54.0" http.response.duration="81.963µs" http.response.status=301 http.response.written=39 instance.id=d8099907-66af-4c26-8797-e7bf994d3bc7 version=v2.6.2 
127.0.0.1 - - [31/Oct/2018:22:01:03 +0000] "GET /v2 HTTP/1.1" 301 39 "" "curl/7.54.0"
time="2018-10-31T22:01:03Z" level=info msg="response completed" go.version=go1.7.6 http.request.host="localhost:5000" http.request.id=cfc8bfaa-bda7-4ace-8466-7e014e21602a http.request.method=GET http.request.remoteaddr="127.0.0.1:60716" http.request.uri="/v2/" http.request.useragent="curl/7.54.0" http.response.contenttype="application/json; charset=utf-8" http.response.duration=1.235065ms http.response.status=200 http.response.written=2 instance.id=d8099907-66af-4c26-8797-e7bf994d3bc7 version=v2.6.2 
127.0.0.1 - - [31/Oct/2018:22:01:03 +0000] "GET /v2/ HTTP/1.1" 200 2 "" "curl/7.54.0"
```

**Comment by Baykonur** [comment]
I have solved this issue once I found out that [Docker for Mac runs inside a VM](https://github.com/docker/for-mac/issues/1160#issuecomment-302908470)

```
$ docker pull busybox
Using default tag: latest
latest: Pulling from library/busybox
90e01955edcd: Pull complete 
Digest: sha256:2a03a6059f21e150ae84b0973863609494aad70f0a80eaeb64bddd8d92465812
Status: Downloaded newer image for busybox:latest
$ docker tag busybox bck.internal.com/busybox:latest
$ docker images
REPOSITORY                 TAG                 IMAGE ID            CREATED             SIZE
bck.internal.com/busybox   latest              59788edf1f3e        4 weeks ago         1.15MB
busybox                    latest              59788edf1f3e        4 weeks ago         1.15MB
$ docker push bck.internal.com/busybox:latest
The push refers to repository [bck.internal.com/busybox]
8a788232037e: Pushed 
latest: digest: sha256:915f390a8912e16d4beb8689720a17348f3f6d1a7b659697df850ab625ea29d5 size: 527
```

However I have another issue with draft:

```
$ draft config set registry bck.internal.com
panic: assignment to entry in nil map

goroutine 1 [running]:
main.(*configSetCmd).run(0xc420237450, 0x0, 0x0)
	/go/src/github.com/Azure/draft/cmd/draft/config_set.go:44 +0x54
main.newConfigSetCmd.func2(0xc42076fb00, 0xc420832d00, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/Azure/draft/cmd/draft/config_set.go:28 +0x2a
github.com/Azure/draft/vendor/github.com/spf13/cobra.(*Command).execute(0xc42076fb00, 0xc420832c60, 0x2, 0x2, 0xc42076fb00, 0xc420832c60)
	/go/src/github.com/Azure/draft/vendor/github.com/spf13/cobra/command.go:698 +0x47a
github.com/Azure/draft/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42076e6c0, 0xc42000c010, 0xc42076e6c0, 0xc420a75f70)
	/go/src/github.com/Azure/draft/vendor/github.com/spf13/cobra/command.go:783 +0x30e
github.com/Azure/draft/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42076e6c0, 0x0, 0x0)
	/go/src/github.com/Azure/draft/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/go/src/github.com/Azure/draft/cmd/draft/draft.go:179 +0x2d
```

Could you please advise?

**Comment by Baykonur** [comment]
Also solved this, before setting the registry, `draft init` should be executed.

**Comment by Baykonur** [comment]
I had another hurdle, although I solved it maybe I should write here so anyone facing the same issue can benefit: My local registry bck.internal.com resolves to a local IP where haproxy listens and balances any request arriving there to the 3 worker-k8s-VMs where I setup [ambassador](https://www.getambassador.io), it is basically my gateway.
First issue I had was that bck.internal.com exists on internet and I had to edit /etc/hosts of my k8s cluster --CentOS (they all connect to internet via my mbp --bridge network). The second issue was that the Docker daemon on the workers were trying to connect to my insecure docker registry bck.internal.com in a secure way (instead of http://bck.internal.com, https://bck.internal.com) so the haproxy was refusing the connections and frontend and backend containers could not be pulled, therefore I had to add bck.internal.com as insecure docker registry on all the VMs and restart the hole cluster.
